app/src/androidTestEreferrals/java/org/eyeseetea/malariacare/rules/PreferencesDependenciesRules.java

```diff
@@ -0,0 +1,148 @@
+import org.eyeseetea.malariacare.BuildConfig;
+import org.eyeseetea.malariacare.data.IAuthenticationDataSource;
+import org.eyeseetea.malariacare.data.authentication.AuthenticationManager;
+import org.eyeseetea.malariacare.data.database.datasources.CountryVersionLocalDataSource;
+import org.eyeseetea.malariacare.data.database.datasources.SettingsDataSource;
+import org.eyeseetea.malariacare.data.database.datasources.SurveyLocalDataSource;
+import org.eyeseetea.malariacare.data.database.datasources.UserAccountDataSource;
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.data.remote.AuthenticationWSDataSource;
+import org.eyeseetea.malariacare.data.remote.IForgotPasswordDataSource;
+import org.eyeseetea.malariacare.data.repositories.OrganisationUnitRepository;
+import org.eyeseetea.malariacare.data.repositories.ProgramRepository;
+import org.eyeseetea.malariacare.data.server.CustomMockServer;
+import org.eyeseetea.malariacare.data.sync.exporter.ConvertToWSVisitor;
+import org.eyeseetea.malariacare.data.sync.exporter.WSPushController;
+import org.eyeseetea.malariacare.data.sync.exporter.eReferralsAPIClient;
+import org.eyeseetea.malariacare.domain.boundary.IConnectivityManager;
+import org.eyeseetea.malariacare.domain.boundary.executors.IAsyncExecutor;
+import org.eyeseetea.malariacare.domain.boundary.executors.IMainExecutor;
+import org.eyeseetea.malariacare.domain.boundary.repositories.IAppInfoRepository;
+import org.eyeseetea.malariacare.domain.boundary.repositories.ICredentialsRepository;
+import org.eyeseetea.malariacare.domain.boundary.repositories.IDeviceRepository;
+import org.eyeseetea.malariacare.domain.boundary.repositories.IInvalidLoginAttemptsRepository;
+import org.eyeseetea.malariacare.domain.boundary.repositories.IOrganisationUnitRepository;
+import org.eyeseetea.malariacare.domain.boundary.repositories.IProgramRepository;
+import org.eyeseetea.malariacare.domain.boundary.repositories.ISettingsRepository;
+import org.eyeseetea.malariacare.domain.boundary.repositories.ISurveyRepository;
+import org.eyeseetea.malariacare.domain.entity.AppInfo;
+import org.eyeseetea.malariacare.domain.entity.Credentials;
+import org.eyeseetea.malariacare.domain.entity.Device;
+import org.eyeseetea.malariacare.domain.entity.InvalidLoginAttempts;
+import org.eyeseetea.malariacare.domain.entity.Program;
+import org.eyeseetea.malariacare.domain.usecase.LoginUseCase;
+import org.eyeseetea.malariacare.domain.usecase.push.PushUseCase;
+import org.eyeseetea.malariacare.domain.usecase.push.SurveysThresholds;
+import org.eyeseetea.malariacare.presentation.executors.UIThreadExecutor;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.mockito.Mock;
+
+import java.io.IOException;
+import java.util.Date;
+
+import static org.mockito.Mockito.when;
+
+public class PreferencesDependenciesRules implements TestRule {
```

Comment:
@idelcano this class has no sense because has many responsibilities:
1 - Mock server logic, It's good create a rule specific for this as in example on this article http://xurxodev.com/favoreciendo-la-composicion-sobre-la-herencia-tambien-en-los-test/
2 - givenALoginUseCase is used only from LoginUseCaseShould, so I think this method  should be where is called -> LoginUseCaseShould
3 - With others for pushusecase and pushcontroller occur the same